File: guns/guns-admin/src/main/java/com/stylefeng/guns/modular/system/model/Setting.java
package com.stylefeng.guns.modular.system.model;

import com.baomidou.mybatisplus.enums.IdType;
import com.baomidou.mybatisplus.annotations.TableId;
import com.baomidou.mybatisplus.annotations.TableField;
import com.baomidou.mybatisplus.activerecord.Model;
import com.baomidou.mybatisplus.annotations.TableName;
import io.swagger.models.auth.In;
import lombok.Getter;
import lombok.Setter;

import java.io.Serializable;

/**
 * <p>
 * 系统设置表
 * </p>
 *
 * @author Arron
 * @since 2019-02-28
 */
@TableName("t_setting")
@Getter
@Setter
public class Setting extends Model<Setting> {

    private static final long serialVersionUID = 1L;

    @TableId(value = "id", type = IdType.AUTO)
    private Integer id;
    /**
     * 官方服务电话号码
     */
    @TableField("service_phone")
    private String servicePhone;
    /**
     * 阿里云云存储access_key
     */
    @TableField("ali_oss_access_key")
    private String aliOssAccessKey;
    /**
     * 阿里云云存储access_id
     */
    @TableField("ali_oss_access_id")
    private String aliOssAccessId;
    /**
     * 阿里云实名认证接口appcode
     */
    @TableField("ali_sm_appcode")
    private String aliSmAppcode;
    /**
     * 云片网(短信)appkey
     */
    @TableField("yp_appkey")
    private String ypAppkey;
    /**
     * 高德地图key
     */
    @TableField("gd_key")
    private String gdKey;
    /**
     * 阿里支付宝appid
     */
    @TableField("ali_pay_app_id")
    private String aliPayAppId;
    /**
     * 阿里支付宝私匙
     */
    @TableField("ali_pay_app_private_key")
    private String aliPayAppPrivateKey;
    /**
     * 阿里支付宝公匙
     */
    @TableField("ali_pay_app_public_key")
    private String aliPayAppPublicKey;
    /**
     * 阿里支付宝回调地址
     */
    @TableField("ali_pay_app_ali_notify_url")
    private String aliPayAppAliNotifyUrl;
    /**
     * 微信支付appid
     */
    @TableField("wechat_pay_app_id")
    private String wechatPayAppId;
    /**
     * 微信支付商户号
     */
    @TableField("wechat_pay_mch_id")
    private String wechatPayMchId;
    /**
     * 微信支付回调地址
     */
    @TableField("wechat_pay_notify_url")
    private String wechatPayNotifyUrl;
    /**
     * 微信支付商户key
     */
    @TableField("wechat_pay_key")
    private String wechatPayKey;
    /**
     * 微信订阅号appid
     */
    @TableField("wechat_app_id")
    private String wechatAppId;
    /**
     * 微信订阅号appsecret
     */
    @TableField("wechat_app_secret")
    private String wechatAppSecret;
    /**
     * 微信订阅号access_token(不用展示)
     */
    @TableField("wechat_access_token")
    private String wechatAccessToken;
    /**
     * 微信订阅号ticket(不用展示)
     */
    @TableField("wechat_ticket")
    private String wechatTicket;
    /**
     * 安卓版本号
     */
    @TableField("android_version")
    private String androidVersion;
    /**
     * 安卓APK下载地址
     */
    @TableField("android_apk_url")
    private String androidApkUrl;

    @TableField("money_give_man")
    private Integer moneyGiveMan;


    @TableField("money_give_women")
    private Integer moneyGiveWomen;

    @TableField("start_get_tel")
    private Integer startGetTel;

    @TableField("publish_switch")
    private String publishSwitch;

    @TableField("remind")
    private String remind;

    @TableField("tel")
    private String tel;

    @Override
    protected Serializable pkVal() {
        return this.id;
    }

    @Override
    public String toString() {
        return "Setting{" +
        "id=" + id +
        ", servicePhone=" + servicePhone +
        ", aliOssAccessKey=" + aliOssAccessKey +
        ", aliOssAccessId=" + aliOssAccessId +
        ", aliSmAppcode=" + aliSmAppcode +
        ", ypAppkey=" + ypAppkey +
        ", gdKey=" + gdKey +
        ", aliPayAppId=" + aliPayAppId +
        ", aliPayAppPrivateKey=" + aliPayAppPrivateKey +
        ", aliPayAppPublicKey=" + aliPayAppPublicKey +
        ", aliPayAppAliNotifyUrl=" + aliPayAppAliNotifyUrl +
        ", wechatPayAppId=" + wechatPayAppId +
        ", wechatPayMchId=" + wechatPayMchId +
        ", wechatPayNotifyUrl=" + wechatPayNotifyUrl +
        ", wechatPayKey=" + wechatPayKey +
        ", wechatAppId=" + wechatAppId +
        ", wechatAppSecret=" + wechatAppSecret +
        ", wechatAccessToken=" + wechatAccessToken +
        ", wechatTicket=" + wechatTicket +
        ", androidVersion=" + androidVersion +
        ", androidApkUrl=" + androidApkUrl +
        "}";
    }
}

File: guns/guns-admin/src/main/webapp/static/modular/setting/setting/setting.js
/**
 * 全局配置管理初始化
 */
var Setting = {
    id: "SettingTable",	//表格id
    seItem: null,		//选中的条目
    table: null,
    layerIndex: -1
};

/**
 * 初始化表格的列
 */
Setting.initColumn = function () {
    return [
        {field: 'selectItem', radio: true},
            //{title: '', field: 'id', visible: true, align: 'center', valign: 'middle'},
            {title: '投诉电话', field: 'servicePhone', visible: true, align: 'center', valign: 'middle'},
            {title: '阿里云云存储access_key', field: 'aliOssAccessKey', visible: true, align: 'center', valign: 'middle'},
            {title: '阿里云云存储access_id', field: 'aliOssAccessId', visible: true, align: 'center', valign: 'middle'},
            {title: '阿里云实名认证接口appcode', field: 'aliSmAppcode', visible: true, align: 'center', valign: 'middle'},
            {title: '云片网(短信)appkey', field: 'ypAppkey', visible: true, align: 'center', valign: 'middle'},
            {title: '高德地图key', field: 'gdKey', visible: true, align: 'center', valign: 'middle'},
            {title: '阿里支付宝appid', field: 'aliPayAppId', visible: true, align: 'center', valign: 'middle'},
            {title: '阿里支付宝私匙', field: 'aliPayAppPrivateKey', visible: true, align: 'center', valign: 'middle'},
            {title: '阿里支付宝公匙', field: 'aliPayAppPublicKey', visible: true, align: 'center', valign: 'middle'},
            {title: '阿里支付宝回调地址', field: 'aliPayAppAliNotifyUrl', visible: true, align: 'center', valign: 'middle'},
            {title: '微信支付appid', field: 'wechatPayAppId', visible: true, align: 'center', valign: 'middle'},
            {title: '微信支付商户号', field: 'wechatPayMchId', visible: true, align: 'center', valign: 'middle'},
            {title: '微信支付回调地址', field: 'wechatPayNotifyUrl', visible: true, align: 'center', valign: 'middle'},
            {title: '微信支付商户key', field: 'wechatPayKey', visible: true, align: 'center', valign: 'middle'},
            {title: '微信服务号appid', field: 'wechatAppId', visible: false, align: 'center', valign: 'middle'},
            {title: '微信服务号appsecret', field: 'wechatAppSecret', visible: false, align: 'center', valign: 'middle'},
            {title: '安卓版本号', field: 'androidVersion', visible: false, align: 'center', valign: 'middle'},
            //{title: '安卓APK下载地址', field: 'androidApkUrl', visible: true, align: 'center', valign: 'middle'}
    ];
};

/**
 * 检查是否选中
 */
Setting.check = function () {
    var selected = $('#' + this.id).bootstrapTable('getSelections');
    if(selected.length == 0){
        Feng.info("请先选中表格中的某一记录！");
        return false;
    }else{
        Setting.seItem = selected[0];
        return true;
    }
};

/**
 * 点击添加全局配置
 */
Setting.openAddSetting = function () {
    var index = layer.open({
        type: 2,
        title: '添加全局配置',
        area: ['100%', '100%'], //宽高
        fix: false, //不固定
        maxmin: true,
        content: Feng.ctxPath + '/setting/setting_add'
    });
    this.layerIndex = index;
};

/**
 * 打开查看全局配置详情
 */
Setting.openSettingDetail = function () {
    if (this.check()) {
        var index = layer.open({
            type: 2,
            title: '全局配置详情',
            area: ['100%', '100%'], //宽高
            fix: false, //不固定
            maxmin: true,
            content: Feng.ctxPath + '/setting/setting_update/' + Setting.seItem.id
        });
        this.layerIndex = index;
    }
};

/**
 * 删除全局配置
 */
Setting.delete = function () {
    if (this.check()) {
        var ajax = new $ax(Feng.ctxPath + "/setting/delete", function (data) {
            Feng.success("删除成功!");
            Setting.table.refresh();
        }, function (data) {
            Feng.error("删除失败!" + data.responseJSON.message + "!");
        });
        ajax.set("settingId",this.seItem.id);
        ajax.start();
    }
};

/**
 * 查询全局配置列表
 */
Setting.search = function () {
    var queryData = {};
    queryData['condition'] = $("#condition").val();
    Setting.table.refresh({query: queryData});
};

$(function () {
    var defaultColunms = Setting.initColumn();
    var table = new BSTable(Setting.id, "/setting/list", defaultColunms);
    table.setPaginationType("client");
    Setting.table = table.init();
});
